internal/configuration/configuration.go

```diff
@@ -69,7 +69,13 @@ func NewConfigurationManager(dataDir string) *Manager {
 	return &mgr
 }
 
-func (m *Manager) RegisterObserver(observer Observer) {
+// RegisterObserver registers an Observer to the config, that will get called
+// when any config changes happens.
+// If withInitConfig is set, observer will be called with the current config.
+func (m *Manager) RegisterObserver(observer Observer, withInitConfig bool) {
```

Comment:
I think managing dependencies between observers might be easier if the initial Update round would be performed at the same time for all of them, after they have been registered.
So the flow would be:
```
configManager.RegisterObserver(a, true)
configManager.RegisterObserver(b, false)
configManager.RegisterObserver(c, true)
// ...
configManager.Update() // or configManager.NotifyObservers() that would call Update on each observier with initialization enabled
```

Reply:
It'll not work, mainly because when workload start, emit some WorkloadStarted messages, and metrics does not have the config, etc.. It's a bit of chicken-egg issue.

Reply:
Understood. :+1: